From 9157b0ad4d016ebc7dd22dc4b50d20fd0c609044 Mon Sep 17 00:00:00 2001
From: Julien Cristau <jcristau@debian.org>
Date: Tue, 6 Apr 2010 11:51:53 +0200
Subject: [PATCH 20/25] XGE: don't register an extension event

The GenericEvent is a core event, we never send an extension event, so
don't reserve an id for one.

Signed-off-by: Julien Cristau <jcristau@debian.org>
Reviewed-by: Peter Hutterer <peter.hutterer@who-t.net>
Signed-off-by: Keith Packard <keithp@keithp.com>
(cherry picked from commit fc5d76740851725e3788c68e14474a012a205892)

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
---
 Xext/geext.c |   13 +------------
 Xext/geint.h |    3 ---
 2 files changed, 1 insertions(+), 15 deletions(-)

diff --git xorg-server-1.8.0/Xext/geext.c xorg-server-1.8.0/Xext/geext.c
index 2ba0ca8..f0a3d87 100644
--- xorg-server-1.8.0/Xext/geext.c
+++ xorg-server-1.8.0/Xext/geext.c
@@ -36,11 +36,8 @@
 
 #define rClient(obj) (clients[CLIENT_ID((obj)->resource)])
 
-int GEEventBase;
-int GEErrorBase;
 static int GEClientPrivateKeyIndex;
 DevPrivateKey GEClientPrivateKey = &GEClientPrivateKeyIndex;
-int GEEventType; /* The opcode for all GenericEvents will have. */
 
 int RT_GECLIENT  = 0;
 
@@ -178,10 +175,6 @@ GEResetProc(ExtensionEntry *extEntry)
 {
     DeleteCallback(&ClientStateCallback, GEClientCallback, 0);
     EventSwapVector[GenericEvent] = NotImplemented;
-
-    GEEventBase = 0;
-    GEErrorBase = 0;
-    GEEventType = 0;
 }
 
 /*  Calls the registered event swap function for the extension.
@@ -225,14 +218,10 @@ GEExtensionInit(void)
     }
 
     if((extEntry = AddExtension(GE_NAME,
-                        GENumberEvents, GENumberErrors,
+                        0, GENumberErrors,
                         ProcGEDispatch, SProcGEDispatch,
                         GEResetProc, StandardMinorOpcode)) != 0)
     {
-        GEEventBase = extEntry->eventBase;
-        GEErrorBase = extEntry->errorBase;
-        GEEventType = GEEventBase;
-
         memset(GEExtensions, 0, sizeof(GEExtensions));
 
         EventSwapVector[GenericEvent] = (EventSwapPtr) SGEGenericEvent;
diff --git xorg-server-1.8.0/Xext/geint.h xorg-server-1.8.0/Xext/geint.h
index 881d306..10a33ce 100644
--- xorg-server-1.8.0/Xext/geint.h
+++ xorg-server-1.8.0/Xext/geint.h
@@ -37,9 +37,6 @@
 #include "extnsionst.h"
 #include <X11/extensions/geproto.h>
 
-extern _X_EXPORT int GEEventType;
-extern _X_EXPORT int GEEventBase;
-extern _X_EXPORT int GEErrorBase;
 extern _X_EXPORT DevPrivateKey GEClientPrivateKey;
 
 typedef struct _GEClientInfo {
-- 
1.7.0.4

